### PR TITLE
Fix workspace resolution for development

### PR DESCRIPTION
**Pull request by restingbull**
Align versions tween bzlmod and workspace
Switch to the fancy bazel ci rules for rbe.
Enable RBE for integration tests (huuuuge difference)

fixes: https://github.com/bazelbuild/rules_kotlin/issues/1230

### CODE REVIEW

File: .bazelrc [hunk 1]
```diff
@@ -3,6 +3,10 @@ common --enable_bzlmod=true
 common:rbe --java_runtime_version=11
 common:rbe --tool_java_runtime_version=11
 
+common:workspace --enable_bzlmod=false --enable_workspace=true
+#common:workspace --enable_platform_specific_config
```
Comment (Bencodes) [inline .bazelrc:7]: These are commented out, is that intentional?

File: src/main/starlark/core/repositories/BUILD [hunk 2]
```diff
@@ -39,7 +39,7 @@ bzl_library(
     name = "repositories",
     srcs = glob(["*.bzl"]),
     visibility = ["//:__subpackages__"],
-    deps = [
-        "@rules_proto//proto:defs",
-    ],
+    #    deps = [
```
Comment (Bencodes) [inline src/main/starlark/core/repositories/BUILD:42]: Did you mean to leave this commented out?

File: .bazelrc (hunk 1 above)
Comment (restingbull) [reply .bazelrc]: Yeah, tried using a hermetic toolchain. It didn't help.

File: src/main/starlark/core/repositories/BUILD (hunk 2 above)
Comment (restingbull) [reply src/main/starlark/core/repositories/BUILD]: Unused dep.

**Review by Bencodes — APPROVED** [review]

**Pull request merged**